CMakeLists.txt

```diff
@@ -68,7 +73,25 @@ endif()
 if(SHPXX_INSTRUMENT_COVERAGE)
     include(AddCoverageTarget)
 endif()
+if(SHPXX_INSTRUMENT_ASAN)
+    add_compile_options(-g3 -fsanitize=address)
+    add_link_options(-fsanitize=address)
+endif()
+if(SHPXX_INSTRUMENT_MSAN)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+        add_compile_options(-g3 -fsanitize=memory)
+        add_link_options(-fsanitize=memory)
+    endif()
+endif()
+if(SHPXX_INSTRUMENT_UBSAN)
+    add_compile_options(-g3 -fsanitize=undefined)
+    add_link_options(-fsanitize=undefined)
+endif()
```

Comment:
The other option for this that I like is to add https://github.com/arsenm/sanitizers-cmake as a submodule